R/colocboost_pipline.R

```diff
@@ -0,0 +1,667 @@
+#' \itemize{
+#'   \item sumstats: A list of summary statistics for the matched LD_info, each sublist contains sumstats, n, var_y from \code{load_rss_data}.
+#'   \item LD_info: A list of LD information, each sublist contains combined_LD_variants, combined_LD_matrix, ref_panel  \code{load_LD_matrix}.
+#' }
+#'
+#' @export
+qc_regional_data <- function(region_data,
+                             # - individual
+                             maf_cutoff = 0.0005, 
+                             pip_cutoff_to_skip_ind = 0,
+                             # - sumstat
+                             skip_region = NULL,
+                             remove_indels = FALSE,
+                             pip_cutoff_to_skip_sumstat = 0,
+                             qc_method = c("rss_qc", "dentist", "slalom"),
+                             impute = TRUE, 
+                             impute_opts = list(rcond = 0.01, R2_threshold = 0.6, minimum_ld = 5, lamb = 0.01)){
+    
+    # - individual level data QC
+    individual_data <- region_data$individual_data
+    if (!is.null(individual_data)){
+        X <- individual_data$residual_X
+        Y <- individual_data$residual_Y
+        MAF <- individual_data$maf
+        # 1. remove maf < maf_cutoff
+        # 2. initial check PIP
+        individual_data <- data_initial_screen_individual(X = X, Y = Y, MAF = MAF, 
+                                                          maf_cutoff = maf_cutoff, 
+                                                          pip_cutoff_to_skip = pip_cutoff_to_skip_ind )
+    }
+    
+    # - summary statistics QC
+    sumstat_data = region_data$sumstat_data
+    if (!is.null(sumstat_data)){
+        # - 1. initial check PIP
+        sumstat_data <- data_initial_screen_sumstat(sumstat_data, skip_region = skip_region,
+                                                    remove_indels = remove_indels, 
+                                                    pip_cutoff_to_skip = pip_cutoff_to_skip_sumstat)
+        
+        # - qc or impute
+        sumstat_data <- sumstat_qc(sumstat_data, qc_method = qc_method,
+                                   impute = impute, impute_opts = impute_opts)
+        
+    }
+    return(list(individual_data = individual_data, 
+                sumstat_data = sumstat_data))
+
+}
+
+
+
+
+
+############# Individual - level data
+#' Add context names to colname of Y if missing
+#' @return A list of residual Y
+#' @noRd
+add_context_to_Y <- function(res_Y){
+    res <- lapply(1:length(res_Y), function(iy){
+        y <- res_Y[[iy]]
+        if (is.null(colnames(y))){ 
+            colnames(y) <- names(res_Y)[iy] 
+        } else {
+            colnames(y) <- paste0(names(res_Y)[iy], "_", colnames(y))
+        }
+        return(y)
+    })
+    names(res) <- names(res_Y)
+    return(res)
+}
+
+#' Remove variants with maf < maf_cutoff
+#' @return A list of residual X and maf
+#' @export
+filter_resX_maf <- function(res_X, maf, maf_cutoff = 0.0005){
```

Comment:
We have this function already: https://github.com/StatFunGen/pecotmr/blob/main/R/misc.R#L163 when MAF is provided it will filter by the provided MAF instead of recomputing it.